Documentation/proxy.md

```diff
@@ -34,4 +34,65 @@ etcd -proxy on -listen-client-urls http://127.0.0.1:8080 -discovery https://disc
 #### Fallback to proxy mode with discovery service
 If you bootstrap a etcd cluster using [discovery service][discovery-service] with more than the expected number of etcd members, the extra etcd processes will fall back to being `readwrite` proxies by default. They will forward the requests to the cluster as described above. For example, if you create a discovery url with `size=5`, and start ten etcd processes using that same discovery url, the result will be a cluster with five etcd members and five proxies. Note that this behaviour can be disabled with the `proxy-fallback` flag.
 
+### Promote a proxy to a member of etcd cluster
+
+A Proxy is not part of the core etcd cluster. A proxy will not promote itself to an etcd member automtically in any case.
+
+If you want to promote a proxy to an etcd member, there are four steps you need to follow:
+
+- add the proxy node as an etcd member into the existing cluster
+- stop the etcd process or service
+- remove the existing proxy data directory
+- restart the etcd process with member configuration
+
+
+#### Example
+
+We assume you have a one member etcd cluster with one proxy. The cluster information is listed below:
+
+|Name|Address|
+|------|---------|
+|infra0|10.0.1.10|
+|proxy0|10.0.1.11|
+
+This example walks you through a case that you promote one proxy to an etcd member. The cluster will become a two member cluster after finishing the four steps.
+
+### Add a new member into the existing cluster
+
+``` bash
+$ etcdctl -endpoint http://10.0.1.10:2379 member add infra1 http://10.0.1.11:2380
+added member 9bf1b35fc7761a23 to cluster
+
+ETCD_NAME="infra1"
+ETCD_INITIAL_CLUSTER="infra0=http://10.0.1.10:2380,infra1=http://10.0.1.11:2380"
+ETCD_INITIAL_CLUSTER_STATE=existing
+```
+
+### Stop the proxy
+
+``` bash
+kill %proxy_pid%
+```
+
+or (if you are running etcd proxy as etcd service under systemd)
+
+``` bash
+systemctl stop etcd2
+```
+
+### Remove the existing data dir
+
+```
+rm -rf %data_dir%/proxy
+```
+
+### Start the etcd as a member
+
+```
+$ export ETCD_NAME="infra1"
+$ export ETCD_INITIAL_CLUSTER="infra0=http://10.0.1.10:2380,infra1=http://10.0.1.11:2380"
+$ export ETCD_INITIAL_CLUSTER_STATE=existing
+$ etcd -listen-client-urls http://10.0.1.11:2379 -advertise-client-urls http://10.0.1.11:2379  -listen-peer-urls http://10.0.1.11:2380 -initial-advertise-peer-urls http://10.0.1.11:2380 -data-dir %data_dir%
+```
+
```

Comment:
> Add a systemd drop-in example to go with the systemd unit from above?

@robszumski I don't think we should add it:
1. etcd doesn't bind with systemd unit tightly
2. systemd unit contains the part to interact with OS, and it may vary depending on local OS. For example, `WantedBy=multi-user.target` in `[Install]` section.
3. people could infer systemd unit from general command line.

> Add some common troubleshooting steps?

@xiang90 I think we could link to error case in https://github.com/coreos/etcd/blob/master/Documentation/runtime-configuration.md#error-cases